go/cmd/dolt/commands/engine/sqlengine.go

```diff
@@ -278,6 +286,67 @@ func configureBinlogReplicaController(config *SqlEngineConfig, engine *gms.Engin
 	return nil
 }
 
+// configureEventScheduler configures the event scheduler with the |engine|.
+func configureEventScheduler(config *SqlEngineConfig, engine *gms.Engine, sessFactory sessionFactory, pro dsqle.DoltDatabaseProvider) error {
+	// need to give correct user, use the definer as user to run the event definition queries
+	ctxFactory := sqlContextFactory()
+
+	// getCtxFunc is used to create new session context for event scheduler.
+	// It starts a transaction that needs to be committed using the function returned.
+	getCtxFunc := func() (*sql.Context, func() error, error) {
+		sess, err := sessFactory(sql.NewBaseSession(), pro)
+		if err != nil {
+			return nil, func() error { return nil }, err
+		}
+
+		newCtx, err := ctxFactory(context.Background(), sess)
+		if err != nil {
+			return nil, func() error { return nil }, err
+		}
+
+		// Starting transaction at the start of the server to load all events to
+		// the event scheduler is causing read replica test to fail as it tries
+		// pulling from remotes for branches that its remoteRef does not exist yet.
+		err = sql.SystemVariables.SetGlobal(dsess.SkipReplicationErrors, true)
+		if err != nil {
+			return nil, func() error { return nil }, err
+		}
+
+		err = sess.SetSessionVariable(newCtx, sql.AutoCommitSessionVar, true)
+		if err != nil {
+			return nil, func() error { return nil }, err
+		}
+
+		// set dolt_show_branch_databases to 'true' to get all revision databases
+		// when accessing all databases. When retrieving events from each database,
+		// only the default branch database will return events that have 'ENABLE' status.
+		// All other databases will return events with 'DISABLE' status regardless of the
+		// stored status.
+		err = sess.SetSessionVariable(newCtx, dsess.ShowBranchDatabases, true)
```

Comment:
I pulled this out temporarily, since none of the existing tests rely on this, but after digging through the code, I think I see why this is in here now. For next step... it would be good to add a test triggering the case where we need this, then looking at if there are any other ways we can filter to only the events on the default branch for event databases.

Reply:
One of the big issues I hit when I turned off `dolt_show_branch_databases` was that events didn't get rescheduled after restarting the sql-server. Now that we are polling for whether events need to be reloaded (to support out-of-band event changes, such as `dolt reset` or `dolt merge`), I didn't want to poll every branch when we know only events from the main branch will ever be scheduled. 

So, I added a new BATS test that asserts that events continue getting scheduled correctly after a sql-server restart and I was able to turn off `dolt_show_branch_databases` with a small tweak to the function that checks if the event is coming from the main branch of an event database.